omnigibson/utils/teleop_utils.py

```diff
@@ -0,0 +1,540 @@
+            self.vr_profile.set_virtual_world_anchor_transform(self.og2xr(robot_base_pos, robot_base_orn[[0, 2, 1, 3]]))
+        # update raw data
+        self._update_devices()
+        self._update_device_transforms()
+        self._update_button_data()
+        if self.use_hand_tracking:
+            self._update_hand_tracking_data()
+        # generate teleop data
+        self.teleop_data["transforms"]["base"] = np.zeros(4)
+        if 1 in self.controllers:
+            self.teleop_data["transforms"]["right"] = (
+                self.raw_data["transforms"]["controllers"][1][0],
+                T.quat_multiply(self.raw_data["transforms"]["controllers"][1][1], self.robot.teleop_rotation_offset[self.robot.arm_names[-1]])
+            )
+            self.teleop_data["transforms"]["base"][0] = self.raw_data["button_data"][1]["axis"]["touchpad_y"] * self.base_movement_speed
+            self.teleop_data["transforms"]["base"][3] = -self.raw_data["button_data"][1]["axis"]["touchpad_x"] * self.base_movement_speed
+            self.teleop_data["gripper_right"] = self.raw_data["button_data"][1]["axis"]["trigger"]
+        if 0 in self.controllers:
+            self.teleop_data["transforms"]["left"] = (
+                self.raw_data["transforms"]["controllers"][0][0],
+                T.quat_multiply(self.raw_data["transforms"]["controllers"][0][1], self.robot.teleop_rotation_offset[self.robot.arm_names[0]])
+            )
+            self.teleop_data["transforms"]["base"][1] = -self.raw_data["button_data"][0]["axis"]["touchpad_x"] * self.base_movement_speed
+            self.teleop_data["transforms"]["base"][2] = self.raw_data["button_data"][0]["axis"]["touchpad_y"] * self.base_movement_speed
+            self.teleop_data["gripper_left"] = self.raw_data["button_data"][0]["axis"]["trigger"]
+        # update robot attachment info
+        if 1 in self.controllers and self.raw_data["button_data"][1]["press"]["grip"]:
+            if not self.reset_button_pressed:
+                self.reset_button_pressed = True
+                self.robot_attached = not self.robot_attached
+        else:
+            self.reset_button_pressed = False
+        self.teleop_data["robot_attached"] = self.robot_attached
+
+    def compute_anchor_offset_with_controller_input(self) -> np.ndarray:
+        """
+        Compute the desired anchor translational offset based on controller touchpad input
+        Returns:
+            np.ndarray: 3d translational offset *in hmd frame*
+        """
+        offset = np.zeros(3)
+        if 1 in self.controllers:
+            right_axis_state = self.controllers[1].get_axis_state()
+            offset = np.array([right_axis_state["touchpad_x"], right_axis_state["touchpad_y"], 0])
+        if 0 in self.controllers:
+            offset[2] = self.controllers[0].get_axis_state()["touchpad_y"]
+        # normalize offset
+        length = np.linalg.norm(offset)
+        if length != 0:
+            offset *= 0.03 / length
```

Comment:
why 0.03? Also, if the length is 0, no normalization is needed? Finally, you should probably check for close to 0 instead of exactly 0.